sklearn/metrics/classification.py

```diff
@@ -240,6 +240,8 @@ def confusion_matrix(y_true, y_pred, labels=None, sample_weight=None):
         labels = unique_labels(y_true, y_pred)
     else:
         labels = np.asarray(labels)
+        if np.all([l not in y_true for l in labels]):
```

Comment:
I'm not sure it will be faster unless I'm doing something stupid:

```
In [1]: import numpy as np

In [2]: x = np.random.randint(0, 5, 1000000)

In [3]: labels = np.array([0, 1, 2, 3, 6])

In [4]: %timeit np.all([l not in x for l in labels])
100 loops, best of 3: 6.38 ms per loop

In [5]: %timeit np.intersect1d(labels, x).size == 0
10 loops, best of 3: 19.4 ms per loop

In [6]: %timeit np.intersect1d(labels, np.unique(x), assume_unique=True).size == 0
10 loops, best of 3: 19.4 ms per loop
```